.github/workflows/javaapi.yml

```diff
@@ -0,0 +1,123 @@
+name: javaapi
+
+on:
+  push:
+    paths-ignore:
+      - "tools/**"
+
+  pull_request:
+    paths-ignore:
+      - "tools/**"
+      - "docs/**"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test_javaapi:
```

Comment:
This workflow is too complex. Too many jobs are for preparing test data.
Why not prepare test data offline and upload them to github?